lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -341,16 +345,12 @@ public function delete_item( $request ) {
 			$result = wp_trash_post( $id );
 		}
 
-		if ( ! $result ) {
+		if ( $result ) {
```

Comment:
We should flip this, and eliminate the unneeded else. In general, the pattern should be like:
```php
if ( ! $x ) {
    return new WP_Error;
}

if ( ! $y ) {
    return new WP_Error;
}

return 'success';
```